### PR TITLE
Convert Akka.Persistence.TestKit.Tests to async

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Still getting deadlocks in here.

**Comment by Arkatufus** [comment]
--blame-crash only available on .NET 5.0, so for .NET FX and .NET core 3.1 we can only use --blame

**Comment by Arkatufus** [comment]
Deadlock fixed

**Comment by Arkatufus** [comment]
Deadlock fixed

File: src/core/Akka.TestKit/TestKitBase.cs [hunk 1]
```diff
@@ -189,6 +188,30 @@ protected void InitializeTest(ActorSystem system, ActorSystemSetup config, strin
             _testState.TestActor = testActor;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void WaitUntilTestActorIsReady(IActorRef testActor)
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestKitBase.cs:192]: Great catch - should we move this to its own PR so we can get a clean signal on this test library?

Comment (Arkatufus) [reply src/core/Akka.TestKit/TestKitBase.cs]: Will do

**Comment by Aaronontheweb** [comment]
Ok, let's see if this gives us a clean signal now - might have to do a rebase if not

**Comment by Aaronontheweb** [comment]
Looks good to me @Arkatufus - onto the next suite
